### PR TITLE
Oj 2467 hmrc transaction ref

### PR DESCRIPTION
**Pull request by mikebeeby**
### What changed

- Retrieved the txn value in the matching lambda when it successfully matches. 
- Modified the logger in the matching lambda to use it's own impl, this is so I could attach the txn as a key. 
- Store txn in the session table.
- Send txn in TxMA audit logs. 

### Why did it change

To log the txn value to TxMA

### Issue tracking

- [OJ-2467](https://govukverify.atlassian.net/browse/OJ-2467)

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

[OJ-2467]: https://govukverify.atlassian.net/browse/OJ-2467?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lambdas/matching/src/matching-handler.ts [hunk 1]
```diff
@@ -29,31 +25,42 @@ export class MatchingHandler implements LambdaInterface {
...
+      });
       throw error;
     }
   }
 }
 
 const handlerClass = new MatchingHandler();
 export const lambdaHandler = handlerClass.handler.bind(handlerClass);
+
+function addLogEntry(event: MatchEvent, txn: string | null, context: Context) {
```
Comment (Caitlin-cooling) [inline lambdas/matching/src/matching-handler.ts:58]: Like this!

File: integration-tests/step-functions/aws/nino_issue_credential/nino_issue_credential.test.ts [hunk 2]
```diff
@@ -452,6 +452,7 @@ describe("Nino Check Hmrc Issue Credential", () => {
     clientSessionId: "252561a2-c6ef-47e7-87ab-93891a2a6a41",
     persistentSessionId: "156714ef-f9df-48c2-ada8-540e7bce44f7",
     evidenceRequest,
+    txn: "2bba5d6e-dc56-4291-b1ed-4f54d263ea63"
```
Comment (Caitlin-cooling) [inline integration-tests/step-functions/aws/nino_issue_credential/nino_issue_credential.test.ts:455]: There's some `txn: expect.any(String),` in this file do you think we could change them to expect a proper value now?

**Review by Caitlin-cooling — APPROVED** [review]
Nice!

File: step-functions/audit_event.asl.json [hunk 3]
```diff
@@ -211,8 +211,34 @@
...
+                },
+                {
+                  "Variable": "$$.Execution.Input.detail.evidence[0].ciReasons",
+                  "IsPresent": true
+                },
+                {
+                  "Not": {
+                    "Variable": "$$.Execution.Input.detail.evidence[0].txn",
+                    "StringEquals": ""
+                  }
```
Comment (jkunle) [inline step-functions/audit_event.asl.json:237]: Why not just?

```
              "Or": [
                {
                  "Not": {
                    "Variable": "$$.Execution.Input.detail.evidence[0].txn",
                    "StringEquals": ""
                  }
```

Comment (jkunle) [reply step-functions/audit_event.asl.json:237]: I think there is a `attemptNum` field `record_check` journey which is currently not in scope. In any case I believe the above is more flexible

Comment (mikebeeby) [reply step-functions/audit_event.asl.json:237]: Hmm I put the other checks in incase the txn was empty on a VC_ISSUED so we did not ignore the rest of the evidence, but I suppose there should always be a txn if a VC has been issued. 
I'll remove the other checks 👍

**Review by jkunle — APPROVED** [review]

**Pull request merged**